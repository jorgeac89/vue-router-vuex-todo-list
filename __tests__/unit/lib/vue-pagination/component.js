import { shallowMount } from '@vue/test-utils';
import VuePagination from 'lib/vue-pagination';

const labels = {
  previousLabel: 'Previous',
  nextLabel: 'Next',
  breakLabel: '...'
};

const cssClasses = {
  containerClass: 'container',
  pageClass: 'page',
  pageLinkClass: 'page-link',
  previousClass: 'previous',
  previousLinkClass: 'previous-link',
  nextClass: 'next',
  nextLinkClass: 'next-link',
  breakClass: 'break',
  breakLinkClass: 'break-link',
  activeClass: 'active',
  activeLinkClass: 'active-link',
  disabledClass: 'disabled'
};

describe('VuePagination component', () => {
  it('renders correctly when the number of pages is 0', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 0,
          forcePage: 0,
          pageCount: 0,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly when there are not enough pages to break the list and the first page is the active one', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 0,
          forcePage: 0,
          pageCount: 2,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly when there are not enough pages to break the list and the last page is the active one', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 1,
          forcePage: 1,
          pageCount: 2,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly when the list should be broken on the right side', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 0,
          forcePage: 0,
          pageCount: 3,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly when the list should be broken on the left side', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 2,
          forcePage: 2,
          pageCount: 3,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly when the list should be broken on both sides', () => {
    expect(
      shallowMount(VuePagination, {
        propsData: {
          ...labels,
          initialPage: 2,
          forcePage: 2,
          pageCount: 5,
          marginPagesDisplayed: 1,
          pageRangeDisplayed: 1,
          onPageChange: jest.fn(),
          ...cssClasses
        }
      })
    ).toMatchSnapshot();
  });
});
