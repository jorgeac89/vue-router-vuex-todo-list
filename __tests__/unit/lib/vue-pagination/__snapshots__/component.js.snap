// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VuePagination component renders correctly when the list should be broken on both sides 1`] = `
<ul class="container">
  <li class="previous"><a aria-disabled="false" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <li class="page"><a aria-label="Page 1" role="button" tabindex="0" class="page-link">
      1
    </a></li>
  <li class="break"><a role="button" tabindex="0" class="break-link">...</a></li>
  <li class="page
      active"><a aria-label="Page 3" role="button" tabindex="0" class="page-link
        active-link">
      3
    </a></li>
  <li class="break"><a role="button" tabindex="0" class="break-link">...</a></li>
  <li class="page"><a aria-label="Page 5" role="button" tabindex="0" class="page-link">
      5
    </a></li>
  <li class="next"><a aria-disabled="false" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;

exports[`VuePagination component renders correctly when the list should be broken on the left side 1`] = `
<ul class="container">
  <li class="previous"><a aria-disabled="false" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <li class="page"><a aria-label="Page 1" role="button" tabindex="0" class="page-link">
      1
    </a></li>
  <li class="break"><a role="button" tabindex="0" class="break-link">...</a></li>
  <li class="page
      active"><a aria-label="Page 3" role="button" tabindex="0" class="page-link
        active-link">
      3
    </a></li>
  <!---->
  <li class="next
      disabled"><a aria-disabled="true" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;

exports[`VuePagination component renders correctly when the list should be broken on the right side 1`] = `
<ul class="container">
  <li class="previous
      disabled"><a aria-disabled="true" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <!---->
  <li class="page
      active"><a aria-label="Page 1" role="button" tabindex="0" class="page-link
        active-link">
      1
    </a></li>
  <li class="break"><a role="button" tabindex="0" class="break-link">...</a></li>
  <li class="page"><a aria-label="Page 3" role="button" tabindex="0" class="page-link">
      3
    </a></li>
  <li class="next"><a aria-disabled="false" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;

exports[`VuePagination component renders correctly when the number of pages is 0 1`] = `
<ul class="container">
  <li class="previous
      disabled"><a aria-disabled="true" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <!---->
  <!---->
  <li class="next
      disabled"><a aria-disabled="false" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;

exports[`VuePagination component renders correctly when there are not enough pages to break the list and the first page is the active one 1`] = `
<ul class="container">
  <li class="previous
      disabled"><a aria-disabled="true" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <!---->
  <li class="page
      active"><a aria-label="Page 1" role="button" tabindex="0" class="page-link
        active-link">
      1
    </a></li>
  <!---->
  <li class="page"><a aria-label="Page 2" role="button" tabindex="0" class="page-link">
      2
    </a></li>
  <li class="next"><a aria-disabled="false" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;

exports[`VuePagination component renders correctly when there are not enough pages to break the list and the last page is the active one 1`] = `
<ul class="container">
  <li class="previous"><a aria-disabled="false" aria-label="Previous page" role="button" tabindex="0" class="previous-link">Previous</a></li>
  <li class="page"><a aria-label="Page 1" role="button" tabindex="0" class="page-link">
      1
    </a></li>
  <!---->
  <li class="page
      active"><a aria-label="Page 2" role="button" tabindex="0" class="page-link
        active-link">
      2
    </a></li>
  <!---->
  <li class="next
      disabled"><a aria-disabled="true" aria-label="Next page" role="button" tabindex="0" class="next-link">Next</a></li>
</ul>
`;
